static/js/protein/protein_protein_interaction_graph.tsx

```diff
@@ -0,0 +1,93 @@
+/**
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Protein-protein interaction graph
+ */
+
+import _ from "lodash";
+import React from "react";
+
+import { drawProteinInteractionGraph } from "./chart";
+import { getProteinInteractionGraphData } from "./data_processing_utils";
+import { InteractingProteinType } from "./page";
+import { MultiLevelInteractionGraphData } from "./types";
+
+type InteractionGraphProps = {
+  centerProteinDcid: string;
+  interactionDataDepth1: InteractingProteinType[];
+};
+
+type InteractionGraphState = {
+  graphData: MultiLevelInteractionGraphData;
+  depth: number;
+  scoreThreshold: number;
+  maxInteractions: number;
+};
+
+const DEFAULTS = {
+  DEPTH: 2,
+  MAX_INTERACTIONS: 4,
+  SCORE_THRESHOLD: 0.4,
+  MISSING_SCORE_FILLER: -1,
+};
+
+export class ProteinProteinInteractionGraph extends React.Component<
+  InteractionGraphProps,
+  InteractionGraphState
+> {
+  constructor(props: InteractionGraphProps) {
+    super(props);
+    this.state = {
+      graphData: null,
+      depth: DEFAULTS.DEPTH,
+      scoreThreshold: DEFAULTS.SCORE_THRESHOLD,
+      maxInteractions: DEFAULTS.MAX_INTERACTIONS,
+    };
+  }
+
+  /**
+   * Perform BFS and (re)draw graph
+   */
+  componentDidUpdate(prevProps: InteractionGraphProps | null): void {
+    if (prevProps !== this.props) {
+      const graphData = getProteinInteractionGraphData(
+        this.props.interactionDataDepth1
+      );
+      this.setState({
+        graphData,
+      });
```

Comment:
I think you would want to return after this. Otherwise, if there was originally graphData retrieved from the previous props, you would end up drawing the chart for the previous props graphData when the code continues past line 74.

Reply:
ahhh you're so smart!! done!